src/scripts/main.js

```diff
@@ -1,3 +1,383 @@
+      }
+      break;
+
+    case 'ArrowLeft':
+      left();
+      startToRestart('restart');
+
+      if (notMove) {
+        newCell();
+      }
+      break;
+  }
+
+  const freeCell = some(rows, '');
+
+  if (!notMove && !freeCell) {
+    let count = 0;
+
+    for (let i = 0; i < FIELD_SIZE - 1; i++) {
+      for (let j = 0; j < FIELD_SIZE - 1; j++) {
+        const currentCell = rows[i].children[j];
+        const downCell = rows[i + 1].children[j];
+        const nextCell = rows[i].children[j + 1];
+
+        if (currentCell.innerText === downCell.innerText
+          || currentCell.innerText === nextCell.innerText) {
+          count++;
+        }
+      }
+    }
+
+    if (!count) {
+      showMessage('lose');
+    }
+  }
+
+  const win = some(rows, '2048');
+
+  if (win) {
+    showMessage('win');
+  }
+});
+
+function clearCell(cell) {
+  cell.innerText = '';
+  cell.className = 'field-cell';
+}
+
+function start() {
+  reset();
+  startGame();
+  showMessage('not');
+}
+
+function reset() {
+  [...rows]
+    .map(row => [...row.children]
+      .map(cell => {
+        cell.innerText = '';
+        cell.className = 'field-cell';
+      }));
+  scoreValue.innerText = 0;
+}
+
+function startValue() {
+  return Math.random() > 0.1 ? 2 : 4;
+}
+
+function addCell(coordX, coordY, value = startValue()) {
+  rows[coordY].children[coordX].innerText = value;
+  rows[coordY].children[coordX].classList.add(`field-cell--${value}`);
+}
+
+function newCell() {
+  randomCell(AMOUNT_CELLS);
+
+  if (rows[y].children[x].innerText === '') {
+    addCell(x, y);
+  } else {
+    newCell();
+  }
+}
+
+function startGame() {
+  randomCell(AMOUNT_CELLS);
+  addCell(x, y);
+  randomCell(AMOUNT_CELLS);
+  addCell(x, y);
+}
+
+function randomCell(amount) {
+  const random = Math.floor(Math.random() * amount);
+
+  if (number !== random) {
+    number = random;
+    x = Math.floor(number / FIELD_SIZE);
+    y = number - (x * FIELD_SIZE);
+  } else {
+    randomCell(amount);
+  }
+}
+
+function startToRestart(selector) {
+  switch (selector) {
+    case 'restart':
+      button.classList.remove('start');
+      button.classList.add('restart');
+      button.innerText = 'Restart';
+      break;
+
+    case 'start':
+      button.classList.add('start');
+      button.classList.remove('restart');
+      button.innerText = 'Start';
+      break;
+  }
+}
+
+function showMessage(typeMessage) {
+  const loseMessage = document.querySelector('.message-lose');
+  const winMessage = document.querySelector('.message-win');
+  const startMessage = document.querySelector('.message-start');
+
+  switch (typeMessage) {
+    case 'lose':
+      winMessage.classList.add('hidden');
+      startMessage.classList.add('hidden');
+      loseMessage.classList.remove('hidden');
+      break;
+
+    case 'win':
+      loseMessage.classList.add('hidden');
+      startMessage.classList.add('hidden');
+      winMessage.classList.remove('hidden');
+      break;
+
+    case 'not':
```

Comment:
consider using a more descriptive name for this case

Reply:
winMessage => showMessageWin?